src/services/exporter.ts

```diff
@@ -0,0 +1,152 @@
+import { Metric, queryByMetric } from '@corpglory/tsdb-kit';
+import { Target } from '../types/target';
+import { URL } from 'url';
+import { apiKeys } from '../config';
+import * as csv from 'fast-csv';
+import * as fs from 'fs';
+import { promisify } from './common';
+import * as moment from 'moment';
+import * as path from 'path';
+
+const MS_IN_DAY = 24 * 60 * 60 * 1000;
+const TIMESTAMP_COLUMN = 'timestamp'
+
+export class Exporter {
+  private exportedRows = 0;
+  private createdTimestamp: number;
+  private user: string;
+
+  private initCsvStream() {
+    const csvStream = csv.createWriteStream({headers: true});
+    const writableStream = fs.createWriteStream(this.getFilePath('csv'));
+
+    csvStream.pipe(writableStream);
+    writableStream.on('finish', async () => {
+      console.log(`Everything is written to ${this.getFilename('csv')}`);
+      await this.updateStatus('finished', 1);
+    })
+
+    return csvStream;
+  }
+
+  public async updateStatus(status: string, progress: number) {
+    try {
+      let time = moment().valueOf();
+      let data = {
+        time,
+        user: this.user,
+        exportedRows: this.exportedRows,
+        progress: progress.toLocaleString('en', {style: 'percent'}),
+        status,
+        datasourceName: 'all',
+      };
+
+      await promisify(fs.writeFile, this.getFilePath('json'), JSON.stringify(data), 'utf8')
+    } catch (err) {
+      console.error(err);
+      throw new Error('Can`t write file');
+    }
+  }
+
+  public async export(data: Target[], user: string, from: number, to: number) {
+    this.user = user;
+
+    this.validateTargets(data);
+    const targets = data.map(target => ({
+      ...target,
+      metric: new Metric(target.datasource, target.targets)
+    }));
+
+    const stream = this.initCsvStream();
+    const days = Math.ceil((to - from) / MS_IN_DAY);
+
+    console.log(`Total days: ${days}`);
+
+    for (let day = 0; day < days; day++) {
+      to = from + MS_IN_DAY;
+
+      console.log(`${day} day: ${from}ms -> ${to}ms`);
+
+      const columns = [TIMESTAMP_COLUMN];
+      const values = {};
+
+      for (const [index, target] of targets.entries()) {
+        const host = new URL(target.panelUrl).origin;
+        const apiKey = apiKeys[host];
+
+        const datasourceMetrics = await queryByMetric(target.metric, target.panelUrl, from, to, apiKey);
+
+        const column = `${target.panelTitle.replace(' ', '-')}` +
+          `-${target.panelId}-${datasourceMetrics.columns[1]}`;
+
+        columns.push(column);
+
+        for (const row of datasourceMetrics.values) {
+          const [timestamp, value] = row;
+
+          if (!values[timestamp]) {
```

Comment:
```suggestion
          if (values[timestamp] !== undefined) {
```

please, let's be more explicit